etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -0,0 +1,220 @@
+            INSERT INTO hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT source_patient_id, health_center, enrollment_date, art_start_date, outcome_date, outcome FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NULL
+            ;
+        ''')
+
+        executeMysql("Load initial enrollments into hivmigration_programs table", '''
+            SET @outcome_transferred_out = (select concept_id from concept where uuid = '3cdd5c02-26fe-102b-80cb-0017a47871b2');
+
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                starting_health_center,
+                enrollment_date,
+                IF (art_start_date < health_center_transfer_date, art_start_date, NULL),
+                health_center_transfer_date,
+                @outcome_transferred_out
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load current enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                health_center,
+                health_center_transfer_date,
+                IF (art_start_date < health_center_transfer_date, health_center_transfer_date, art_start_date),
+                outcome_date,
+                outcome
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load to patient_program table", '''
+            SET @hiv_program = (SELECT program_id FROM program WHERE uuid = "b1cb1fc1-5190-4f7a-af08-48870975dafc");
+            
+            insert into patient_program
+                (patient_program_id, patient_id, program_id, date_enrolled, date_completed, location_id, outcome_concept_id, creator, date_created, uuid)
+            select
+                h.patient_program_id, 
+                p.person_id,
+                @hiv_program,
+                h.enrollment_date, 
+                h.outcome_date, 
+                hc.openmrs_id,
+                h.outcome 
+                1,
+                date_format(curdate(), '%Y-%m-%d %T'),
+                uuid()
+            from 
+                hivmigration_programs h 
+            inner join 
+                hivmigration_patients p on h.source_patient_id = p.source_patient_id 
+            left outer join
+                hivmigration_health_center hc on h.location_id = hc.hiv_emr_id
+            ; 
+        ''')
+        
+        // TODO: figure out how patient state should work
```

Comment:
Indeed.  The main thing is that we are not handling art start date.  Perhaps we are not intending to but then we don't need to select it into the staging table and use it at all.  What this was targeting was a model like we have in Malawi, where the patient is enrolled in the HIV Program, and they have states within that program, one of which is "On Antiretrovirals", and the start date of that state is their ART start date.  This may or may not be what we want to do.